website/docs/more/debugging.mdx

```diff
@@ -4,40 +4,57 @@ title: 'Debugging'
-`panic!`s and outputs them to the console. Yew will automatically catch `panic!`s and log them to
-your browser's console.
+Yew automatically logs panics in the browser console.
 
 ## Console Logging
 
-In general, Wasm web apps are able to interact with Browser APIs, and the `console.log` API is no
-exception. There are a few options available:
+In JavaScript, `console.log()` is used to log to the browser console. There are two main options for Yew, both of which can log any `JsValue`.
 
 ### [`wasm-logger`](https://crates.io/crates/wasm-logger)
 
-This crate integrates with the familiar Rust `log` crate:
+This crate integrates with the [`log`](https://crates.io/crates/log) crate.
+`wasm_logger` sends the log level, source line and filename to the browser console.
 
 ```rust ,ignore
-// setup
+use log::info;
+use wasm_bindgen::JsValue;
+
 fn main() {
     wasm_logger::init(wasm_logger::Config::default());
+
+    let object = JsValue::from("world");
+    info!("Hello {}", object.as_string().unwrap());
 }
+```
+
+### [`gloo-console`](https://crates.io/crates/gloo-console)
+
+This crate is part of Gloo, a collection of libraries providing ergonomic Rust wrappers for browser APIs.
+The `log!` macro can take a `JsValue` directly which is slightly easier to use than `wasm_logger`.
 
-// usage
-log::info!("Update: {:?}", msg);
+```rust ,ignore
+use gloo_console::log;
+use wasm_bindgen::JsValue;
+
+fn main() {
+    let object = JsValue::from("world");
+    log!("Hello", object)
+}
 ```
 
 ## Debugging component lifecycles
 
-Yew makes use of the [`tracing`](https://crates.io/crates/tracing) crate to emit debug information about lifecycle events of components. These can be used to trace why and when a component is created, updated, rerendered or destroyed.
+[`tracing`](https://crates.io/crates/tracing) can be used to collect event information related to a component's lifecycle. `tracing` also comes with a feature flag for `log` support, which integrates nicely with `wasm-logger`.
 
-To make use of this, you need to [register a `Subscriber`](https://docs.rs/tracing/latest/tracing/#in-executables) or enable the `"log"` feature for compatibility with the `log` crate, e.g. when using `wasm-logger`. A subscriber is supposed to be chosen only by the final executable crate, not by library crates. You can also turn on compile-time [flags to disable logging](https://docs.rs/tracing/latest/tracing/level_filters/index.html#compile-time-filters) statically, for a smaller wasm file size.
+[Compile time filters](https://docs.rs/tracing/latest/tracing/level_filters/index.html#compile-time-filters) can be used to adjust verbosity or disable logging, which should result in a smaller Wasm file.
 
 ## Source Maps
 
-There is currently no first-class support for source maps for Rust / Wasm web apps. This, of course, is subject to change. If this is no longer true or if progress is made, please suggest a change!
+Source maps are currently not supported.
```

Comment:
[Chrome can display DWARF source maps](https://developer.chrome.com/blog/wasm-debugging-2019/#enter-dwarf)

I haven't tested it so it's better to test it before documenting it though

Reply:
I use it sometimes, it *basically* works, but has some inconvenience with debugging code in `std` and other library crates where you have to manually add some configuration (and keep it up to date with cargo versions :/). Definitely a bit fiddly to setup and it's a shame that it doesn't exist (or work) in Firefox.